cmd/dusk/server.go

```diff
@@ -271,7 +273,17 @@ func (s *Server) OnConnection(conn net.Conn, addr string) {
 		log.Panic(err)
 	}
 
-	go peer.Create(context.Background(), peerReader, peerWriter, writeQueueChan)
+	ctx, cancel := context.WithCancel(context.Background())
+	errChan := peer.Create(ctx, peerReader, peerWriter, writeQueueChan)
+	go handlePeerError(errChan, cancel)
```

Comment:
Returning channels is not very idiomatic in go. IMO, a more readable version would be:

```
func (s *Server) OnAccept(conn net.Conn) {
    ...
    go func() {
        ctx, cancel := context.WithCancel(context.Background())
        defer cancel()
        err := peer.Create(ctx, peerReader, peerWriter, writeQueueChan)
        logServer.WithError(err).Warnln("error received from peer")
    }
    ...
}

func Create(ctx context.Context, reader *Reader, writer *Writer, writeQueueChan <-chan bytes.Buffer) chan error {
        errChan := make(chan error, 1)
        defer close(errChan)

	go reader.ReadLoop(ctx, errChan)
	go writer.Serve(ctx, writeQueueChan, errChan)

        return <-errChan
}
```

Reply:
Just changed it to that :)